weather_aplication/ViewController.swift

```diff
@@ -1,25 +1,86 @@
-//
-//  ViewController.swift
-//  weather_aplication
-//
-//  Created by Adam Głąbicki Airnauts on 29/07/2022.
-//
 import SnapKit
 import UIKit
 
 class ViewController: UIViewController {
-
+    private let kLogoWidth: CGFloat = 250
+    private let kLogoHeight: CGFloat = 250
+    private let kLeftMargin: CGFloat = 10
+    private let kRightMargin: CGFloat = 10
+    private let kTopMargin: CGFloat = 20
+    private let kBottomMargin: CGFloat = 20
+    private let backgroundView = UIView()
+    private let aplicationImage = UIImage(named: "weather_symbol.png")
+    private let logoImageView = UIImageView()
+    private let aplicationNameLabel = UILabel()
+    private let proceedButton = UIButton()
+    private let aplicationNameString: String = "MyWeather"
+    private let buttonString: String = "Proceed"
+    
     override func viewDidLoad() {
         super.viewDidLoad()
         // Do any additional setup after loading the view.
         setupView()
+        setupConstraints()
         
+        proceedButton.addTarget(self, action: #selector(proceedButtonPressed), for: .touchUpInside)
+    }
+    @objc func proceedButtonPressed(){
+        let mainViewController = MainViewControler()
+        navigationController?.pushViewController(mainViewController, animated: true)
     }
+    
     func setupView(){
         view.backgroundColor = .blue
+        
+        backgroundView.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(backgroundView)
+        
+        logoImageView.translatesAutoresizingMaskIntoConstraints = false
+        logoImageView.image = aplicationImage
+        backgroundView.addSubview(logoImageView)
+        
+        aplicationNameLabel.text = aplicationNameString
+        aplicationNameLabel.textColor = .white
+        aplicationNameLabel.textAlignment = .center
+        aplicationNameLabel.backgroundColor = .black
+        aplicationNameLabel.translatesAutoresizingMaskIntoConstraints = false
+        backgroundView.addSubview(aplicationNameLabel)
+        
+        proceedButton.translatesAutoresizingMaskIntoConstraints = false
+        proceedButton.setTitle(buttonString, for: .normal)
+        proceedButton.setTitleColor(.red, for: .normal)
+        proceedButton.backgroundColor = .yellow
+        view.addSubview(proceedButton)
     }
     
     func setupConstraints(){
+        backgroundView.snp.makeConstraints{make in
+            make.top.equalTo(view.snp.topMargin).offset(kTopMargin)
+            make.left.equalTo(view.snp.leftMargin).offset(kLeftMargin)
+            make.right.equalTo(view.snp.rightMargin).offset(-kRightMargin)
+            make.centerY.equalToSuperview()
+            make.centerX.equalToSuperview()
```

Comment:
Coś z tymi constraintami jest nie tak. Skoro w linijce 36 dodałeś _backgroundView_ jako subview do _view_, to tutaj wystarczy że odniesiesz się do superview zamiast view.snp.xxx. Poza tym nie za bardzo rozumiem do czego potrzebny jest w ogóle backgroundView, skoro chcemy mieć tylko 3 obiekty- zdjęcie, labelkę i guzik.